src/test_solvers/solvers.cu

```diff
@@ -0,0 +1,137 @@
+#include <iostream>
+
+#include "solvers.h"
+#include <cusp/monitor.h>
+#include <cusp/print.h>
+#include <cusp/krylov/gmres.h>
+
+#include <cuda_runtime.h>
+// where to perform the computation
+typedef cusp::device_memory MemorySpace;
+
+std::vector<float> cudaSolver(std::vector<int>* rowInd, std::vector<int>* colInd, std::vector<float>* val, std::vector<float>* b){
```

Comment:
I don't know exactly if `rowInd` or `colInd` are modified, but just doing pattern matching it sounds like the cudaMemcpy copies the data to the device, so maybe the input actually can be const ? So a `const std::vector<int>& rowInd`.

Since I don't know anything about CUDA, I am most certainly wrong, so if so, please disregard.

Reply:
Great point! I will remember to use "const" often in future work.